src/Cassandra/HostConnectionPool.cs

```diff
@@ -128,15 +131,27 @@ public HostConnectionPool(Host host, Configuration config, Serializer serializer
         /// Gets an open connection from the host pool (creating if necessary).
         /// It returns null if the load balancing policy didn't allow connections to this host.
         /// </summary>
+        /// <exception cref="DriverInternalError" />
+        /// <exception cref="BusyPoolException" />
+        /// <exception cref="UnsupportedProtocolVersionException" />
+        /// <exception cref="SocketException" />
+        /// <exception cref="AuthenticationException" />
         public async Task<Connection> BorrowConnection()
         {
             var connections = await EnsureCreate().ConfigureAwait(false);
             if (connections.Length == 0)
             {
                 throw new DriverInternalError("No connection could be borrowed");
             }
-            var c = MinInFlight(connections, ref _connectionIndex, _maxInflightThreshold);
-            ConsiderResizingPool(c.InFlight);
+
+            var c = MinInFlight(connections, ref _connectionIndex, _maxRequestsPerConnection, out var inFlight);
```

Comment:
There's now (unintended?) interplay between `_maxRequestsPerConnection` and `_maxInflightThresholdToConsiderResizing` because this use to return a connection if it was less than `_maxInflightThreshold` (default: 128) and now it returns a connection if it's less than `_maxRequestsPerConnection` (default: 2048) and I think this could potentially cause `ConsiderResizingPool()` to create connections sooner that it would have before when considering multiple connections.

Reply:
This might not be a big deal.

Reply:
I haven't thought of that, good thinking!

As the load should be fairly balanced between different connections to the same host, it's unlikely that  connections are so unbalanced that it causes a creation of new connection, it should only happen when there are 3 or more connections and 2 of them are above `_maxInflightThreshold`. Having the driver to create a new connection in that case is fine by me, I agree its not a big deal.